internal/agent-api/client.go

```diff
@@ -171,6 +171,10 @@ func (a *AgentClient) Drain() error {
 			// no-op for now, try the next one... perhaps we should return the error here in the future?
 		}
 
+		for sub.IsDraining() {
+			time.Sleep(time.Millisecond * 10)
```

Comment:
where did this value come from? Is it always 10 millis or does it change depending on the machine?

Reply:
Currently it's just hard-coded to 10 millis. Since draining is async this blocks until we know the sub has drained. An optimization would be to use a waitgroup to drain concurrently but seemed like a premature optimization.